swap/src/monero.rs

```diff
@@ -99,6 +99,31 @@ impl Amount {
         self.0
     }
 
+    pub fn as_monero_decimal(&self) -> Decimal {
+        let piconero_decimal = Decimal::from(self.0);
+        let pic_offcet_decimal = Self::piconero_offset_decimal();
+        piconero_decimal / pic_offcet_decimal
+    }
+
+    fn piconero_offset_decimal() -> Decimal {
+        Decimal::from(PICONERO_OFFSET)
+    }
+
+    pub fn max_bitcoin_from_balance(&self, satoshis: u64) -> Result<bitcoin::Amount> {
+        let piconero_minus_fee = self.as_piconero().saturating_sub(MONERO_FEE.as_piconero());
+
+        if piconero_minus_fee == 0 {
+            tracing::error!("The current Monero balance is smaller than the basic locking fee");
+            bail!("The current Monero balance is smaller than the basic locking fee");
```

Comment:
Either log or return error please but not both! That will lead to duplicate information in the logs :)

Reply:
Done!